### ISSUE TITLE
Dokka with Kotlin 1.6.20-RC2 can't resolve dependency: Error class

### ISSUE THREAD

**Issue by hfhbd** [Kotlin/dokka#2411]
**Describe the bug**
I use Kotlin 1.6.20-RC2 with `multiplatform` and a common dependency.

**Expected behaviour**
Documentation referencing the external class from the dependency is resolved

**Screenshots**
<img width="487" alt="Bildschirmfoto 2022-03-29 um 19 39 40" src="https://user-images.githubusercontent.com/22521688/160672362-1058a6b3-fa9e-4a42-84ac-6d89bb7ee811.png">

**To Reproduce**
[untitled1.zip](https://github.com/Kotlin/dokka/files/8373820/untitled1.zip)

**Dokka configuration**
```Kotlin
plugins {
    kotlin("multiplatform") version "1.6.20-RC2"
    id("org.jetbrains.dokka") version "1.6.10"
}

repositories {
    mavenCentral()
}

kotlin {
    js(IR) {
        browser()
    }
    jvm()

    sourceSets {
        commonMain {
            dependencies {
                // 0.0.12 uses Kotlin 1.5.30, 0.0.13-RC uses 1.6.20-RC2
                implementation("app.softwork:kotlinx-uuid-core:0.0.13-RC")
            }
        }
    }
}
```

**Installation**
- Operating system: macOS 12
- Build tool: Gradle v7.4.1
- Dokka version: 1.6.10

**Additional context**
Using Kotlin 1.6.10 works

**Are you willing to provide a PR?**
Sure, but I don't know if this is a Dokka or Kotlin bug

**Comment by IgnatBeresnev** [comment]
Hi! This is indeed a problem, most likely due to compatibility between KMP and dokka, thanks for reporting it! We'll do our best to get it fixed in 1.6.20

**Comment by IgnatBeresnev** [comment]
Will be released in `1.6.21` soon
